File: Behavioral/Lab21/Typescript/src/program.ts

import { AbstractStrategy } from './Strategies/abstract_strategy';
import { AddApartmentStrategy } from './Strategies/add_apartment_strategy';
import { AddHouseStrategy } from './Strategies/add_house_strategy';
import { AddRoomStrategy } from './Strategies/add_room_strategy';
class Program {
    private static _addStrategy : AbstractStrategy
    static main(args: string[]): void {
        console.log("Start testing... \n\n");
        [   new AddApartmentStrategy(),
            new AddHouseStrategy("house", "house", "1", "1", "test"),
            new AddRoomStrategy(),]
        .forEach((strategy: AbstractStrategy) => {
            this._addStrategy = strategy;
            this.executeCommands();
        });
        console.log("End testing... \n\n");
    }

    private static executeCommands(): void {
        console.log(this._addStrategy.execute());
    }
}

Program.main([]);

File: Behavioral/Lab21/C#/Program.cs
using System;
using System.Diagnostics;
using System.Net.WebSockets;
using System.Reflection.Emit;
using System.Runtime.InteropServices;
using C_.Builders;
using C_.TemplateMethods;
using C_.Directors;
using C_.Entities;

namespace C_
{    
    class Program
    {
        private static AbstractStrategy _addStrategy;
        static void Main(string[] args)
        {
            Console.WriteLine("Start testing... \n\n");
            foreach (AbstractStrategy strategy in new List<AbstractStrategy>{
                new AddApartmentStrategy(),
                new AddHouseStrategy("house", "house", "1", "1", "test"),
                new AddRoomStrategy(),})
            {
                _addStrategy = strategy;
                AddEstate();
            };
            Console.WriteLine("End testing... \n\n");
        }

        private static void AddEstate(){
            Console.WriteLine(_addStrategy.Execute());
        }

    }
}
